prac_09/taxi_simulator.py

```diff
@@ -0,0 +1,47 @@
+from prac_09.taxi import Taxi
+from prac_09.silver_service_taxi import SilverServiceTaxi
+
+MENU = "q)uit, c)hoose taxi, d)rive"
+
+
+def main():
+    taxis = [Taxi("Prius", 100), SilverServiceTaxi("Limo", 100, 2), SilverServiceTaxi("Hummer", 200, 4)]
+    current_taxi = None
+    total_fare = 0
+    print("Lets drive!")
+    print(MENU)
+    menu_choice = input(">>> ").upper()
+    while menu_choice != 'Q':
+        if menu_choice == 'C':
+            print("Taxi's available:")
+            display_available_taxis(taxis)
+            taxi_choice = int(input("Choose Taxi: "))
+            try:
+                current_taxi = taxis[taxi_choice]
+            except IndexError:
+                print("Invalid Taxi Choice")
+        elif menu_choice == 'D':
+            if current_taxi:
+                current_taxi.start_fare()
+                distance = float(input("Drive how far? "))
+                current_taxi.drive(distance)
+                fare = current_taxi.get_fare()
+                print(f"Your {current_taxi.name} trip cost ${fare:.2f}")
+                total_fare += fare
+            else:
+                print("You need to choose a taxi before you can drive")
```

Comment:
This seems to work as you need it. This is just a suggestion, I have my if and else statement the other way around. I say "if current_taxi is None:" then "You need to choose a taxi ...". My else statement does everything else. It's not a huge difference, however, it could help to increase the readability of your code and remove the awkward "if current_taxi" statement you have?